### PR TITLE
New Data Source: `azurerm_app_service`

### PR DESCRIPTION
**Pull request by tombuildsstuff**
Adding a new Data Source for App Service, since there's use cases for referencing it (e.g. adding a slot / binding a certificate etc)

### CODE REVIEW

File: azurerm/data_source_app_service_test.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+data "azurerm_app_service" "test" {
+  name                = "${azurerm_app_service.test.name}"
+  resource_group_name = "${azurerm_app_service.test.resource_group_name}"
+}
+`, config)
+}
+
+func testAccDataSourceAppService_siteConfig(rInt int, location string) string {
+	config := testAccAzureRMAppService_32Bit(rInt, location)
```
Comment (katbyte) [inline azurerm/data_source_app_service_test.go]: Is there a reason for the change of the test name from `testAccAzureRMAppService_32Bit` -> `testAccDataSourceAppService_siteConfig`?

File: website/docs/d/app_service.html.markdown [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+`site_config` supports the following:
+
+* `always_on` - Should the app be loaded at all times?
+
+* `default_documents` - The ordering of default documents to load, if an address isn't specified.
+
+* `dotnet_framework_version` - The version of the .net framework's CLR used in this App Service.
+
+* `java_version` - The version of Java to use.
```
Comment (katbyte) [inline website/docs/d/app_service.html.markdown]: I think this should be the "Version of Java in use"

File: website/docs/d/app_service.html.markdown [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+* `java_container` - The Java Container to use.
```
Comment (katbyte) [inline website/docs/d/app_service.html.markdown]: Same as above: "The java container in use"

File: website/docs/d/app_service.html.markdown [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+
+* `java_container_version` - The version of the Java Container to use.
```
Comment (katbyte) [inline website/docs/d/app_service.html.markdown]: Same as above: "The version of the java container in use"

File: website/docs/d/app_service.html.markdown [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+* `python_version` - The version of Python to use in this App Service.
+
+* `remote_debugging_enabled` - Is Remote Debugging Enabled?
+
+* `remote_debugging_version` - Which version of Visual Studio should the Remote Debugger be compatible with?
+
+* `scm_type` - The type of Source Control enabled for this App Service.
+
+* `use_32_bit_worker_process` - Should the App Service run in 32 bit mode, rather than 64 bit mode?
```
Comment (katbyte) [inline website/docs/d/app_service.html.markdown]: Tense seems incorrect for a datasource.

File: website/docs/d/app_service.html.markdown [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+* `websockets_enabled` - Should WebSockets be enabled?
```
Comment (katbyte) [inline website/docs/d/app_service.html.markdown]: Tense seems incorrect for a datasource.

**Comment by tombuildsstuff** [comment]
Tests pass (from the previous commit):

```
$ acctests azurerm TestAccDataSourceAzureRMAppService_
=== RUN   TestAccDataSourceAzureRMAppService_basic
--- PASS: TestAccDataSourceAzureRMAppService_basic (253.82s)
=== RUN   TestAccDataSourceAzureRMAppService_tags
--- PASS: TestAccDataSourceAzureRMAppService_tags (111.16s)
=== RUN   TestAccDataSourceAzureRMAppService_clientAppAffinityDisabled
--- PASS: TestAccDataSourceAzureRMAppService_clientAppAffinityDisabled (103.34s)
=== RUN   TestAccDataSourceAzureRMAppService_siteConfig
--- PASS: TestAccDataSourceAzureRMAppService_siteConfig (107.17s)
=== RUN   TestAccDataSourceAzureRMAppService_appSettings
--- PASS: TestAccDataSourceAzureRMAppService_appSettings (143.14s)
=== RUN   TestAccDataSourceAzureRMAppService_connectionString
--- PASS: TestAccDataSourceAzureRMAppService_connectionString (125.71s)
PASS
ok      github.com/terraform-providers/terraform-provider-azurerm/azurerm       844.388s
```

**Pull request merged**